pkg/cluster/internal/kubeconfig/internal/kubeconfig/merge.go

```diff
@@ -98,5 +98,14 @@ func merge(existing, kind *Config) error {
 	// set the current context
 	existing.CurrentContext = kind.CurrentContext
 
+	// TODO: We should not need this, but it allows broken clients that depend
+	// on apiVersion and kind to work. Notably the upstream javascript client.
+	// See: https://github.com/kubernetes-sigs/kind/issues/1242
+	if len(existing.OtherFields) == 0 {
```

Comment:
We might need a future enhancement specifically to default `"kind"` and `"apiVersion"` but this is simpler, and solves the immediate concern (when empty or otherwise unset existing files lack any of these fields).

Reply:
it also doesn't involve any magical constants.